internal/provider/common/role.go

```diff
@@ -126,17 +126,17 @@ func ValidateWorkspaceDeploymentRoles(ctx context.Context, diags diag.Diagnostic
 	workspaceIds = lo.Intersect(lo.Uniq(workspaceIds), lo.Uniq(deploymentWorkspaceIds))
 	if len(workspaceIds) != len(deploymentWorkspaceIds) {
 		tflog.Error(ctx, "failed to mutate Team roles", map[string]interface{}{"error": err})
-		diags.AddError(
+		return diag.Diagnostics{diag.NewErrorDiagnostic(
 			"Unable to mutate Team roles, not every deployment role has a corresponding workspace role",
 			"Please ensure that every deployment role has a corresponding workspace role",
-		)
-		return diags
+		),
+		}
 	}
-	return diags
+	return nil
 }
 
-// CheckDuplicateWorkspaceId checks if there are duplicate workspace ids in the workspace roles
-func CheckDuplicateWorkspaceId(workspaceRoles []iam.WorkspaceRole) []string {
+// HasDuplicateWorkspaceId checks if there are duplicate workspace ids in the workspace roles
+func HasDuplicateWorkspaceId(workspaceRoles []iam.WorkspaceRole) []string {
```

Comment:
since we're actually still returning the list
```suggestion
func GetDuplicateWorkspaceIds(workspaceRoles []iam.WorkspaceRole) []string {
```